src/extension/extension.ts

```diff
@@ -13,62 +13,341 @@
+			kaotoStatusBarItem.show();
+		}
+	}));
+	context.subscriptions.push(vscode.workspace.onDidCloseTextDocument((event) => {
+		if (event.fileName.endsWith('.camel.yaml')) {
+			kaotoStatusBarItem.hide();
+		}
+	}));
+
+	// create a Port Manager for Camel JBang Run with Dev console
+	const portManager = new PortManager(10111, 10999); // TODO revisit port numbers range
+
+	/*
+	* register integrations view provider
+	*/
+	const integrationsProvider = new IntegrationsProvider();
+	const integrationsTreeView = vscode.window.createTreeView('kaoto.integrations', {
+		treeDataProvider: integrationsProvider,
+		showCollapseAll: true
+	});
+	context.subscriptions.push(integrationsTreeView);
+	context.subscriptions.push(vscode.commands.registerCommand(KAOTO_INTEGRATIONS_VIEW_REFRESH_COMMAND_ID, () => integrationsProvider.refresh()));
+
+	context.subscriptions.push(vscode.commands.registerCommand('kaoto.integrations.editSource', async (integrationEntry: Integration) => {
+		await vscode.window.showTextDocument(vscode.Uri.parse(integrationEntry.filepath));
+	}));
+	context.subscriptions.push(vscode.commands.registerCommand('kaoto.integrations.deleteEntry', async (integrationEntry: Integration) => {
+		const confirmation = await confirmFileDeleteDialog(integrationEntry.description as string); // integrationEntry.description ==> at the moment points to File name
+		if (confirmation) {
+			await vscode.workspace.fs.delete(vscode.Uri.file(integrationEntry.filepath), { useTrash: true });
+			integrationsProvider.refresh();
+			await vscode.window.showInformationMessage(`File '${integrationEntry.description}' was moved to Trash.`);
+		}
+	}));
+	context.subscriptions.push(vscode.commands.registerCommand('kaoto.integrations.jbang.run', async function (integrationEntry: Integration) {
+		if (!vscode.workspace.workspaceFolders) {
+			await vscode.window.showWarningMessage(WORKSPACE_WARNING_MESSAGE);
+			return;
+		}
+		await new CamelRunJBangTask(path.basename(integrationEntry.filepath), path.dirname(integrationEntry.filepath), portManager.allocatePort()).executeOnly();
+		await new Promise((time) => setTimeout(time, 2_500)); // TODO remove static time, at the moment just to give some more time to ensure Camel JBang reflects properly new state
+		deploymentsProvider.refresh();
+		await sendCommandTrackingEvent('kaoto.integrations.jbang.run');
+	}));
+	context.subscriptions.push(vscode.commands.registerCommand('kaoto.integrations.jbang.run.all', async function () {
+		if (!vscode.workspace.workspaceFolders) {
+			await vscode.window.showWarningMessage(WORKSPACE_WARNING_MESSAGE);
+			return;
+		}
+		await new CamelRunJBangTask('*', undefined, portManager.allocatePort()).executeOnly();
+		await new Promise((time) => setTimeout(time, 2_500)); // TODO remove static time, at the moment just to give some more time to ensure Camel JBang reflects properly new state
+		deploymentsProvider.refresh();
+		await sendCommandTrackingEvent('kaoto.integrations.jbang.run.all');
+	}));
```

Comment:
currently the Run All is running only all integrations at the root folder. The othe rintegrations which are displayed in the view whcih are in subfolder are not started